### ISSUE TITLE
Support scipy < 1.11

### ISSUE THREAD

**Issue by abiasiol** [xarray-contrib/flox#317]
Hi Deepak,
Thank you for your amazing work on this package! It is a joy to use it.

---
I found an issue introduced in version `0.8.6`, from [this commit](https://github.com/xarray-contrib/flox/commit/15abf492d3ecc7bb20b29609917ef30a8b7b970a).

Example:
```
import xarray as xr
data = xr.DataArray(data=[1, 2, 3, 4, 5, 6, 7, 8, 9], dims=["time"]).chunk({"time": 2})
grouper = xr.DataArray(data=[1, 1, 1, 2, 2, 1, 1, 1, 1], dims=["time"]).chunk({"time": 2})
data.groupby(grouper).mean()

AttributeError: 'numpy.ndarray' object has no attribute 'tocsr'
```
---
I believe this line [here](https://github.com/xarray-contrib/flox/blob/676c0f03f46ba4d23a49172e9c7a9aa9c0453851/flox/core.py#L331C18-L331C18) is the culprit, where there is an operation between a sparse matrix and a dense numpy array.

I have worked on a fix for that line, but I'm not sure it extends to all cases. I would appreciate your feedback on this possible edit:
```
from scipy.sparse import diags
....
containment = ((asfloat.T @ asfloat) @ diags(1 / chunks_per_label)).tocsr()
``` 
---
After the fix:
```
import xarray as xr
data = xr.DataArray(data=[1, 2, 3, 4, 5, 6, 7, 8, 9], dims=["time"]).chunk({"time": 2})
grouper = xr.DataArray(data=[1, 1, 1, 2, 2, 1, 1, 1, 1], dims=["time"]).chunk({"time": 2})
data.groupby(grouper).mean().compute()

<xarray.DataArray (group: 2)>
array([5.14285714, 4.5       ])
Coordinates:
  * group    (group) int64 1 2
```

**Comment by dcherian** [comment]
Thanks for the kind words :)

Unfortunately I can't reproduce on `main` or on `v0.8.6`. Can you tell me your scipy version please/

**Comment by abiasiol** [comment]
Ah! I was on scipy `1.10.1`. It is solved when using scipy > `1.11.1`. Thank you!

**Comment by dcherian** [comment]
Thanks for the report, that was an oversight!
